ote_sdk/ote_sdk/tests/entities/shapes/test_rectangle.py

```diff
@@ -0,0 +1,673 @@
+        return {"x1": 0.1, "y1": 0.0, "x2": 0.4, "y2": 0.2}
+
+    def horizontal_rectangle(self) -> Rectangle:
+        self.now = now()
+        return Rectangle(**self.horizontal_rectangle_params())
+
+    def vertical_rectangle_params(self) -> dict:
+        return {
+            "x1": 0.1,
+            "y1": 0.1,
+            "x2": 0.3,
+            "y2": 0.4,
+            "labels": self.rectangle_labels(),
+            "modification_date": datetime(
+                year=2020, month=1, day=1, hour=9, minute=30, second=15, microsecond=2
+            ),
+        }
+
+    def vertical_rectangle(self) -> Rectangle:
+        return Rectangle(**self.vertical_rectangle_params())
+
+    @staticmethod
+    def square_params() -> dict:
+        return {"x1": 0.1, "y1": 0.1, "x2": 0.3, "y2": 0.3}
+
+    def square(self) -> Rectangle:
+        self.now = now()
+        return Rectangle(**self.square_params())
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_rectangle_required_parameters(self):
+        """
+        <b>Description:</b>
+        Check Rectangle class instance required parameters
+
+        <b>Input data:</b>
+        Rectangle class initiation parameters
+
+        <b>Expected results:</b>
+        Test passes if Rectangle instance has expected attributes specified during required parameters initiation
+
+        <b>Steps</b>
+        1. Compare x1, y1, x2, y2 and type Rectangle instance parameters with expected values
+        """
+        rectangle = self.horizontal_rectangle()
+        assert rectangle.x1 == 0.1
+        assert rectangle.y1 == 0.0
+        assert rectangle.x2 == 0.4
+        assert rectangle.y2 == 0.2
+        assert rectangle.type == ShapeType.RECTANGLE
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_rectangle_optional_parameters(self):
+        """
+        <b>Description:</b>
+        Check Rectangle optional parameters
+
+        <b>Input data:</b>
+        Instance of Rectangle class
+
+        <b>Expected results:</b>
+        Test passes if Rectangle instance has expected labels and modification attributes specified during Rectangle
+        class object initiation with optional parameters
+
+        <b>Steps</b>
+        1. Compare default label Rectangle instance attribute with expected value
+        2. Compare default modification_date Rectangle instance attribute with expected value
+        3. Compare specified label Rectangle instance attribute with expected value
+        4. Compare specified modification_date Rectangle instance attribute with expected value
+        """
+        # Checking default values of optional parameters
+        default_params_rectangle = self.horizontal_rectangle()
+        assert default_params_rectangle._labels == []
+        assert default_params_rectangle.modification_date == self.now
```

Comment:
it still might fail. call `now()` once, and use it for object initialization and then check equality

Reply:
Thanks. In tests with default values for optional parameters, expected that if parameter modification_date equal “None” (default value) during Rectangle object initialization, then date_time will be equal now(), based on the condition on line 63 of ote_sdk / ote_sdk / entities / shapes / rectangle.py module. In default_parameters tests it is expected that modification_date parameter will not be set manually, there are performance issues on Jenkins, so I used replace microseconds method

Reply:
Could you advice if there is a better way than replacing microseconds and seconds to tests with not specified modification_date parameter for Rectangle instances?

Reply:
> Thanks. In tests with default values for optional parameters, expected that if parameter modification_date equal “None” (default value) during Rectangle object initialization, then now() will be used to set date_time class attribute, based on the condition on line 63 of ote_sdk / ote_sdk / entities / shapes / rectangle.py module. In default_parameters tests it is expected that modification_date parameter will not be set manually, there are performance issues on Jenkins, so I used replace microseconds method for date_time object

The problem is that fist call of `now()` can give us 10h59m59s999999ms, and second call can give us 11h00m00s000001ms, that's why setting microseconds to 0 does not work always.

Reply:
> Could you advice if there is a better way than replacing microseconds and seconds to tests with not specified modification_date parameter for Rectangle instances? Perhaps, in such scenarios, it would be enough to check that modification_date attribute changed its type from None to datetime?

Yes, I think it would be enough to check `type` of `modification_date`

Reply:
Thanks, changed to checking default modification_date attribute “type” instead of “value” for Rectangle instances with non-specified modification_date parameter